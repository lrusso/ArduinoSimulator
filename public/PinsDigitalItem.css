:root {
  --form-control-color: #0078d4;
  --form-control-disabled: #555455;
  --foreground: #959495;
  --accent-color: #0283e7;
}

*,
*:before,
*:after {
  font-family: system-ui, sans-serif;
  box-sizing: border-box;
  color: var(--foreground);
}

body {
  margin: 0;
  padding: 0;
}

form {
  display: flex;
  align-items: baseline;
  margin: 2rem;
  gap: 50px;
}

label {
  font-size: 1.0rem;
}

label input {
  font-size: 1.0rem;
}

.form-control  { 
  font-weight: bold;
  line-height: 1.1;
  display: grid;
  grid-template-columns: auto auto;
  gap: 0.2em;
}

.form-control--disabled {
  color: var(--form-control-disabled);
  cursor: not-allowed;
}

input[type="checkbox"] {
  -webkit-appearance: none;
  appearance: none;
  background-color: #2c303a;
  margin: 0; 
  color: var(--foreground);
  width: 1.15em;
  height: 1.15em;
  border: 0.1em solid #959495;
  border-radius: 0.15em;
  transform: translateY(-0.075em);
  display: grid;
  place-content: center;
}

input[type="checkbox"]::before {
  content: "";
  width: 0.65em;
  height: 0.65em;
  transform: scale(0);
  transition: 120ms transform ease-in-out;
  box-shadow: inset 1em 1em var(--accent-color);
  transform: scale(0);
}

input[type="checkbox"]:checked::before {
  transform: scale(1);
}

input[type="checkbox"]:disabled {
  --form-control-color: var(--form-control-disabled);
  border-radius: 50%;
  background: #2c303a;
  box-shadow: inset #111 0 -1px 8px;
  border: 0.1em solid #555;
  color: var(--form-control-disabled);
  cursor: not-allowed;
  transition: all 0.2s ease;
}

input[type="checkbox"]:disabled::before {
  transform: scale(0);
}

input[type="checkbox"]:checked:disabled {
  background: #24E0FF;
  border: 0.15em solid #555;
  box-shadow: rgba(0, 0, 0, 0.2) 0 -1px 2px 1px, inset #006 0 0 5px, #3F8CFF 0 0 8px;
}

input::-webkit-outer-spin-button,
input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
input[type=number] {
  -moz-appearance: textfield;
}

input[type="number"] {
  width: 5rem;
  background: #2c303a;
  font-size: 1.2rem;
  border: 0.15em solid #959495;
  border-radius: 5px;
  padding-right: 10px;
  color: var(--accent-color);
  text-align: right;
  font-weight: bold;
}

input[type="number"]:focus {
    outline: none; 
}

input[type="number"]:disabled {
  width: 5rem;
  background: #2b2f3925;  
  border: 0.15em solid var(--form-control-disabled);
  color: var(--form-control-disabled);
  border-radius: 3rem;
  cursor: not-allowed;
}