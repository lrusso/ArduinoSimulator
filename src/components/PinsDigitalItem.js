import React from "react"

const PinsDigitalItem = ({ pinNumber, isEnabled }) => {
  return (
    <div
      style={{ backgroundColor: isEnabled ? "green" : "red", ...styles.container }}
    >
      {pinNumber}
    </div>
  )
}

const styles = {
  container: {
    float: "left",
    width: "16px",
    height: "16px",
    borderRadius: "16px",
    marginTop: "4px",
    marginLeft: "4px",
    fontSize: "11px",
    fontFamily: "Arial",
    color: "white",
    justifyContent: "center",
    alignItems: "center",
    display: "flex",
    cursor: "default",
  },
}

export default PinsDigitalItem
