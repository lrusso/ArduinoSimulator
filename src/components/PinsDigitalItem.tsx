import React from "react"

import { Gpio } from "../utils/interfaces"

interface PinsDigitalItemProps {
  gpio: Gpio
}

const PinsDigitalItem = ({ gpio }: PinsDigitalItemProps) => {
  return (
    <div
      style={{
        backgroundColor: gpio.isEnabled ? "green" : "red",
        ...styles.container,
      }}
    >
      {gpio.pinNumber}
    </div>
  )
}

const styles: { [key: string]: React.CSSProperties } = {
  container: {
    float: "left",
    width: "16px",
    height: "16px",
    borderRadius: "16px",
    marginTop: "4px",
    marginLeft: "4px",
    fontSize: "11px",
    fontFamily: "Arial",
    color: "white",
    justifyContent: "center",
    alignItems: "center",
    display: "flex",
    cursor: "default",
  },
}

export default PinsDigitalItem
