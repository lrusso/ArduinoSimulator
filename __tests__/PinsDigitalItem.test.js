import React from "react"
import renderer from "react-test-renderer"
import PinsDigitalItem from "../src/components/PinsDigitalItem"

describe("PinsDigitalItem", () => {
  test("Rendering 1", () => {
    const tree = renderer
      .create(<PinsDigitalItem pinNumber={0} isEnabled={false} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test("Rendering 2", () => {
    const tree = renderer
      .create(<PinsDigitalItem pinNumber={0} isEnabled={true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
