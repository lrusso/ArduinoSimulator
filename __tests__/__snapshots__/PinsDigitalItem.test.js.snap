// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinsDigitalItem Rendering 1 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderRadius": "16px",
      "color": "white",
      "cursor": "default",
      "display": "flex",
      "float": "left",
      "fontFamily": "Arial",
      "fontSize": "11px",
      "height": "16px",
      "justifyContent": "center",
      "marginLeft": "4px",
      "marginTop": "4px",
      "width": "16px",
    }
  }
>
  0
</div>
`;

exports[`PinsDigitalItem Rendering 2 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderRadius": "16px",
      "color": "white",
      "cursor": "default",
      "display": "flex",
      "float": "left",
      "fontFamily": "Arial",
      "fontSize": "11px",
      "height": "16px",
      "justifyContent": "center",
      "marginLeft": "4px",
      "marginTop": "4px",
      "width": "16px",
    }
  }
>
  0
</div>
`;
